tools/schemacode/schema.py

```diff
@@ -26,6 +30,34 @@ def _get_entry_name(path):
         return path.name
 
 
+def search_structure(struct, path):
+    """Recursively search a dictionary-like object for $ref keys.
+
+    Each $ref key is replaced with the contents of the referenced file.
+    """
+    if isinstance(struct, dict):
+        if "$ref" in struct.keys():
+            with open(os.path.join(path, struct["$ref"]), "r") as fo:
+                temp = yaml.load(fo, Loader=yaml.SafeLoader)
+
+            struct.pop("$ref")
+
+            for k_temp, v_temp in temp.items():
+                if k_temp not in struct.keys():
+                    struct[k_temp] = v_temp
```

Comment:
Not a big fan of "temp" variables. And we can do more idiomatic Python for the override code.

```suggestion
                template = yaml.load(fo, Loader=yaml.SafeLoader)

            struct.pop("$ref")
            # Result is template object with local overrides
            struct = {**template, **struct}
```